tests/tensorlake/test_graph_behaviours.py

```diff
@@ -327,14 +328,15 @@ def raise_if_called_with_multiple_values(x: int, y: int, z: int) -> int:
     raise Exception("Should not be called")
 
 
-@tensorlake_router()
-def route_multiple_values(
-    x: int, y: int, z: int
-) -> List[Union[sum_multiple_values, raise_if_called_with_multiple_values]]:
+@tensorlake_function(next=[sum_multiple_values, raise_if_called_with_multiple_values])
+def route_multiple_values(x: int, y: int, z: int) -> RouteTo[
+    Tuple[int, int, int],
+    Union[sum_multiple_values, raise_if_called_with_multiple_values],
+]:
     if x + y + z == 0:
-        return raise_if_called_with_multiple_values
+        return RouteTo((x, y, z), raise_if_called_with_multiple_values)
     else:
-        return sum_multiple_values
+        return RouteTo((x, y, z), sum_multiple_values)
```

Comment:
What would be the error message/UX for the customer if we return a wrong function here (not listed in next=[])

Reply:
Currently, I think it'll just invoke the wrong function(s).  We might want to catch that and do something.